deals-summary/README.md

```diff
@@ -0,0 +1,92 @@
+
+- Run npm install to install dependencies for this project.
+- Create a file named `.env` inside the app.functions folder (inside `src/app`)
+- In your HubSpot account, click on the CRM development item in the main navigation
+- In the sidebar under Tools select Private apps and find the Deals summary app. Click View access token and copy this token to your clipboard.
+- Add it to your `.env` file
+
+![image](https://git.hubteam.com/storage/user/1895/files/8afc64c7-ff3c-4a0d-90bd-542de3a75f5a)
+
+- To begin the dev process and automatically see your changes reflected in the CRM, run hs project dev.
+- Return to the contact record and refresh the browser. You should now see a developing locally icon next to the card title.
+
+## Add the Average Margin component
+
+- Open the [get-data.js](./src/app/app.functions/get-data.js) file in your local text editor. This file is using the [HubSpot API client](https://developers.hubspot.com/docs/api/overview) to fetch deal data related to whichever contact record an end user is viewing in the CRM. This fetching is being done by the `getAssociatedDeals` function. The result of this function is stored in a deals variable. The `calculateTotalAmounts` function is using this data to calculate the total amount of deals.
+- To begin the dev process and automatically see your changes reflected in the CRM, run `hs project dev`.
+- Add a function called `calculateAverageAmount` below the `calculateTotalAmounts` function. It should look like this:
+
+```javascript
+function calculateAverageAmount(deals) {
+  const totalCount = deals.length;
+
+  const amounts = deals.map((deal) => parseFloat(deal.properties.amount));
+  const totalDeals = amounts.reduce((sum, amount) => sum + amount, 0);
+
+  const average = Math.ceil(totalDeals / totalCount);
+  return -Math.round(-average);
+}
+```
+
+- Next add a variable in the exports statement at the top of the file to store the result of this function and add this variable to the send response.
+
+```javascript
+const avgAmount = calculateAverageAmount(deals);
+sendResponse({ deals, totalAmount, avgAmount });
+```
+
+- This is everything required for fetching the data and making it available to the front end. Open the [card-frontend.jsx](./src/app/extensions/card-frontend.jsx) file to add this data to the CRM card.
+- First, add a state variable to define the initial state of the average amount
+
+```javascript
+const [avgAmount, setAvgAmount] = useState(0);
+```
+
+- Next, update the state within the `useEffect` function
+
+```javascript
+setAvgAmount(serverlessResponse.response.avgAmount);
+```
+
+- Finally, add a new statisticsItem component to display the average amount in the CRM card
+
+```html
+<StatisticsItem label="AVG MARGIN" number="{avgAmount}">
+  <Text>Low End</Text>
+</StatisticsItem>
+```
```

Comment:
This should probably be `jsx` instead of `html` and I believe the number prop should be `number={avgAmount}` (without quotes).